### PR TITLE
Use prometheus-alert-buffer-client-gem for alert collection

### PR DESCRIPTION
**Pull request by zgalor**
emove internal message_buffer_client and use prometheus-alert-buffer-client gem to connect and get alerts from prometheus

bz: https://bugzilla.redhat.com/show_bug.cgi?id=1488037

repo for prometheus-alert-buffer-client: https://github.com/openshift/prometheus-alert-buffer-client-ruby

### CODE REVIEW

**Comment by zgalor** [comment]
@moolitayer PTAL

**Comment by zgalor** [comment]
@zeari PTAL

File: app/models/manageiq/providers/kubernetes/monitoring_manager/event_catcher/stream.rb [hunk 1]
```diff
@@ -42,7 +38,7 @@ def fetch
     #   ...
     #   ]
     # }
-    alert_list = response.body
+    alert_list = @ems.connect.get(generation_id => @current_generation, from_index => @current_index)
```
Comment (zeari) [inline app/models/manageiq/providers/kubernetes/monitoring_manager/event_catcher/stream.rb]: Shouldnt it be **:generation_id** => @current_generation

Comment (zgalor) [reply app/models/manageiq/providers/kubernetes/monitoring_manager/event_catcher/stream.rb]: Yes. will fix

File: app/models/manageiq/providers/kubernetes/monitoring_manager_mixin.rb [hunk 2]
```diff
@@ -41,12 +42,16 @@ def verify_credentials(_auth_type = nil, _options = {})
...
-      :verify_ssl => options[:verify_ssl] || verify_ssl,
-      :cert_store => options[:cert_store] || ssl_cert_store
+      :url         => "https://#{prometheus_alerts_endpoint.hostname}:#{prometheus_alerts_endpoint.port}",
+      :path        => "/topics/alerts",
+      :credentials => {:token => authentication_token},
+      :ssl         => {:verify         => verify_ssl,
+                       :ssl_cert_store => ssl_cert_store},
+      :request     => {:open_timeout => settings.open_timeout.to_f_with_method,
+                       :timeout      => settings.timeout.to_f_with_method},
+      :proxy       => options[:http_proxy] || VMDB::Util.http_proxy_uri
```
Comment (moolitayer) [inline app/models/manageiq/providers/kubernetes/monitoring_manager_mixin.rb]: @enoodle do we have all the levels needed for this setting (e.g system wide vs per provider)

Comment (enoodle) [reply app/models/manageiq/providers/kubernetes/monitoring_manager_mixin.rb]: @moolitayer Yes

Comment (moolitayer) [reply app/models/manageiq/providers/kubernetes/monitoring_manager_mixin.rb]: what about the HTTP Proxy we have per provider in the add provider screen, do we use that?

Comment (enoodle) [reply app/models/manageiq/providers/kubernetes/monitoring_manager_mixin.rb]: @moolitayer good catch, In the `ContainerManagerMixin` we send the proxy to `raw_connect` (that later calls this) the value from the provider's `options` [1], here it should be added the same then. that `|| VMDB::Util.http_proxy_uri` is later added before sending the options to the client but there is no real need in this separation.

[1] https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/master/app/models/manageiq/providers/kubernetes/container_manager_mixin.rb#L126

**Comment by zgalor** [comment]
@yaacov @zeari @enoodle @moolitayer please approve after changes

File: spec/vcr_cassettes/manageiq/providers/kubernetes/monitoring_manager.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 http_interactions:
 - request:
     method: get
-    uri: https://alerts-prometheus.example.com/topics/alerts
+    uri: https://alerts-prometheus.example.com/topics/alerts?fromIndex&generationID
```
Comment (moolitayer) [inline spec/vcr_cassettes/manageiq/providers/kubernetes/monitoring_manager.yml]: Eventually please avoid sending these if they are not defined

**Comment by zgalor** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 9a5c69cd1a7c7c2438b9e2438e806de419526249
Author: Mooli Tayer <mtayer@redhat.com>
Date:   Sun Dec 3 13:19:58 2017 +0200

    Merge pull request #175 from zgalor/use_alertbuffer_client
    
    Use prometheus-alert-buffer-client-gem for alert collection
    (cherry picked from commit 3fabdc1e259e0ac7ff09a324d75ba10c3e138de4)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1520491
```

**Pull request merged**